discovery/syncer.go

```diff
@@ -1433,14 +1434,25 @@ func (g *GossipSyncer) FilterGossipMsgs(msgs ...msgWithSenders) {
 	endTime := startTime.Add(
 		time.Duration(g.remoteUpdateHorizon.TimestampRange) * time.Second,
 	)
+
+	var (
+		startBlock tlv.RecordT[tlv.TlvType2, uint32]
+		endBlock   tlv.RecordT[tlv.TlvType4, uint32]
+	)
+	startBlock = g.remoteUpdateHorizon.FirstBlockHeight.UnwrapOr(startBlock)
```

Comment:
Perhaps we just want to make an `UnwrapOrV`? Similar to `WhenSomeV` it will call `Val` for you, so then we cna just pass in the two timestamps above. 

We may also want consider making a type alias for `tlv.RecordT[tlv.TlvType2, uint32]`, so then it can just be `lnwire.ChanUpdStartBlock` (or something like that).